halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/NodeFilter.java

```diff
@@ -64,17 +64,37 @@ public NodeFilter withAnyMaster() {
     return this;
   }
 
-  public static NodeFilter makeRejectAllFilter() {
-    return new NodeFilter();
+  /**
+   * Modifies the current filter to accept anything the reference does.
+   *
+   * @param reference is the reference to accept.
+   */
+  public NodeFilter extendReference(NodeReference reference) {
+    halconfigFile = !reference.halconfigFile.isEmpty() ? reference.halconfigFile : halconfigFile;
+    deployment = !reference.deployment.isEmpty() ? reference.deployment : deployment;
```

Comment:
My point was more about syntactic sugar for the pattern here than the specific syntax itself.
Is there a simplifying Java notation you can apply here?